File: docs/examples/type-example-4.js
import { PointI32, PointI64 } from './type-example-4.zig';

const p1 = new PointI32({ x: 45, y: -12 });
console.log(p1.Type.name, p1.valueOf());
const p2 = new PointI64({ x: 45, y: -12 });
console.log(p2.Type.name, p2.valueOf());

File: docs/examples/type-example-4.zig
fn Point(comptime T: type) type {
    return struct {
        x: T,
        y: T,
        comptime Type: type = T,
    };
}

pub const PointI32 = Point(i32);
pub const PointI64 = Point(i64);
pub const PointF32 = Point(f32);
pub const PointF64 = Point(f64);
